app/models/person.rb

```diff
@@ -108,17 +108,21 @@ def previous_role_appointments
     @previous_role_appointments ||= role_appointments(current: false)
   end
 
+  def previous_non_ministerial_role_appointments
+    @previous_non_ministerial_role_appointments ||=
+      sort_by_appointment_order(
+        all_role_appointments
+          .filter { |role_appointment| role_appointment.dig("details", "ended_on").present? }
+          .filter { |role_appointment| non_ministerial_role?(role_appointment) },
+      )
+  end
 
-  def previous_roles
-    previous_role_appointments
-      .sort_by { |role_appointment| role_appointment["details"]["started_on"] }
-      .reverse
-      .map do |role_appointment|
-        role_appointment["links"]["role"].first.tap do |role|
-          role["start_year"] = Time.zone.parse(role_appointment["details"]["started_on"]).strftime("%Y")
-          role["end_year"] = Time.zone.parse(role_appointment["details"]["ended_on"]).strftime("%Y")
-        end
-      end
+  def formatted_previous_non_ministerial_roles
```

Comment:
Suggested change for your [commit](https://github.com/alphagov/collections/pull/4016/commits/61627f7f2a5eb32af5ba0709e4388b047bf6b99d) **Add support for previous non-ministerial roles**

I think this method name should be shortened to `previous_non_ministerial_roles` as it's only the date being formatted which is pretty minor. It will make it easier to understand the difference between the three methods:

- previous_non_ministerial_roles
- previous_non_ministerial_roles_items
- previous_non_ministerial_role_appointments

Reply:
I think `previous_non_ministerial_roles` is used on line 111 so I don't think I can use it again how about `previous_non_ministerial_roles_with_dates`?